src/components/ViewPCOL/Documents/DocumentView.tsx

```diff
@@ -93,41 +126,129 @@ export const DocumentView = (props: {
-                  <br />
-                  {props.document.Name}
-                </DialogContent>
-                <DialogActions>
+          <>
+            {props.document.ListItemAllFields.DocGroup !== "PCOL" && (
+              <>
+                <Dialog modalType="alert">
                   <DialogTrigger disableButtonEnhancement>
-                    <Button appearance="secondary">Cancel</Button>
+                    <Tooltip
+                      withArrow
+                      content="Edit metadata"
+                      relationship="label"
+                    >
+                      <Button
+                        appearance="transparent"
+                        icon={
+                          editDocument.isPending ? <Spinner /> : <EditIcon />
+                        }
+                        aria-label="Delete"
+                        disabled={editDocument.isPending}
+                      />
+                    </Tooltip>
                   </DialogTrigger>
+                  <DialogSurface>
+                    <DialogBody>
+                      <DialogTitle>Edit document metadata</DialogTitle>
+                      <DialogContent>
+                        <form
+                          style={{
+                            display: "flex",
+                            flexDirection: "column",
+                            width: "100%",
+                          }}
+                        >
+                          <Label htmlFor="FileName">File Name</Label>
+                          <Input
+                            id="FileName"
+                            contentAfter={"." + fileExt}
+                            value={filename}
+                            onChange={(_ev, data) => setFilename(data.value)}
+                          />
+                          <Label htmlFor="DocGroup">Document Group</Label>
+                          <Combobox
+                            value={docGroup[0]}
+                            selectedOptions={docGroup}
+                            onOptionSelect={(_ev, data) => {
+                              setDocGroup(data.selectedOptions);
+                            }}
+                          >
+                            <Option>Attachment</Option>
+                            <Option>Support</Option>
+                          </Combobox>
+                        </form>
+                      </DialogContent>
+                      <DialogActions>
+                        <DialogTrigger disableButtonEnhancement>
+                          <Button appearance="secondary">Cancel</Button>
+                        </DialogTrigger>
+                        <DialogTrigger disableButtonEnhancement>
+                          <Button
+                            appearance="primary"
+                            disabled={filename.length === 0}
+                            onClick={() =>
+                              editDocument.mutate({
```

Comment:
UI gets no feedback on how this update went. For instance, if you exceed 256 characters in the name, or 400 in the full path, SharePoint throws the error on the RestAPI, but we don't see it on the page. Same is true for Delete.